specs/IsSmartScreenRequired

```diff
@@ -0,0 +1,94 @@
+
+[Edge SmartScreen](https://docs.microsoft.com/en-us/deployedge/microsoft-edge-security-smartscreen) helps you identify reported phishing and malware websites, and also helps you make informed decisions about downloads.
+
+Currently, end developers can use `options->put_AdditionalBrowserArguments(L"--disable-features=msSmartScreenProtection")` to control SmartScreen in the WebView2 application. It is essentially a startup parameter of the browser process. It must be determined when the WebView2Environment is created. And and cannot be modified at runtime.
+
+To support more flexibility we introduce a new API.
+We initially considered an API like `CoreWebView2Environment.IsSmartScreenEnabled` that would directly change the value for all the processes. The problem is this is not easy to do for apps like Office who have multiple apps connected to the same browser process. In their case each app has IsSmartScreenEnabled and its hard for the browser process to know which change to the property should win. 
+
+Instead we have CoreWebView2Settings.IsSmartScreenRequired. Each WebView2 declares if it requires SmartScreen. Some WebView2s may be used to display app content and don't require SmartScreen and others may be rendering arbitrary web content and do need SmartScreen. Having SmartScreen on unnecessarily for app content is a detriment to performance but otherwise not a problem. Having SmartScreen off for arbitrary web content is an issue. We have to turn SmartScreen on or off for all the WebView2s in an environment so if any WebView2 requires SmartScreen then we turn it on for all of them. If WebView2 settings change or WebView2s are closed and then all the WebView2s in an environment don't require SmartScreen, then we can turn SmartScreen off.
+
+It is much easier to indicate if individual WebView2s require SmartScreen than to have an end developer explicitly manage if SmartScreen should be enabled as a whole, especially when its different sets of WebView2s in different processes (like Excel's set of WebView2s and Word's set of WebView2s) all sharing the same user data folder.
+In this document we describe the new setting. We'd appreciate your feedback.
+
+
+# Description
+You can use CoreWebView2Settings.IsSmartScreen to control SmartScreen. SmartScreen is enabled or disabled per browser process, so all WebView2 applications sharing the same user data folder path also share SmartScreen being enabled or disabled.
+If CoreWebView2Setting.IsSmartScreenRequired is true for any CoreWebView2 in its associated CoreWebView2Environment, then SmartScreen is enabled. If CoreWebView2Setting.IsSmartScreenRequired is false for all CoreWebView2s in their CoreWebView2Environment, then SmartScreen is disabled.
+The default value for `IsSmartScreenRequired` is true if SmartScreen is enabled when the CoreWebView2 is created and false if SmartScreen is disabled when the CoreWebView2 is created. The value doesn't change if SmartScreen is enabled or disabled later.
```

Comment:
I think this default is dangerous, and it should just default to `true` always.

As written, the default is dependent on state the app does not directly control. (SmartScreen might have been disabled by a partner app that happens to be using the same user data folder.)

For example. Excel creates the first CoreWebView2, and it's using it to show a trusted local image gallery, so it disables SmartScreen for that CoreWebView2. Since that is the only CoreWebView2 so far, this disables SmartScreen for the environment.

Next, Word creates a CoreWebView2 to show a user-submitted Q&A that it thinks will help the user with their problem. Since SmartScreen happens to be disabled in the environment's at the moment, Word's CoreWebView2 is created with SmartScreen not required. The user clicks a malicious link in one of the community-submitted replies to the question and is sent to a phishing site.

The "inherit current global state" algorithm means that every app would have to explicitly set `IsSmartScreenRequired = true` to make sure it's enabled on their CoreWebView2, regardless of what other apps may be doing. In which case, why have a default at all?

This is also a security vulnerability if Excel and Word update at different times. In the above example, suppose Excel 2024 is the one who set `IsSmartScreenRequired = false`. But Word is Word 2021, which predates the `IsSmartScreenRequired` property, so it doesn't even know that it inherited an insecure configuration. Upgrading Excel to 2024 made Word 2021 insecure, which seems bad.

Reply:
Yes let's make the default just `true`.

I was thinking when I read this the first time that defaulting to the current state of SmartScreen was required to avoid the end dev wants SmartScreen off, but every time they create a CoreWebView2 SmartScreen is briefly turned back on, before the end dev changes from the default of true to the false they want. But that's not a problem because the new value only applies on the next navigation or download so the end dev has time to change the property before it is applied.

Defaulting to `true` instead of the current state of SmartScreen is easier to explain and understand.

Reply:
test

Reply:
You've updated the documentation to say "When a new CoreWebview2 
  /// is created, the SmartScreens of all CoreWebviews using the same user data folder are reset to true.", is that right? Can we wait until the CoreWebView2 first navigates before applying the `IsReputationCheckingRequired` property value?

Reply:
@david-risney 

Before a new webview is navigated for the first time, if its smartscreen is not set, all webviews will be reset to true after navigation.

During this time, other webview navigation will not change the value of smartscreen.

If that's what you mean, it's not that hard to do. I'll rework the documentation later.